test/cycles/eulerian.jl

```diff
@@ -16,6 +16,23 @@
         iszero(ne(graph)) && return true
         return false
     end
+
+    function check_trail(original_graph, trail)
+        
+        graph = copy(original_graph)
+        length(trail) < 2 && return true
```

Comment:
You need to check that the graph is empty before returning true. For instance if `eulerian_trail` only returns empty list, then all your tests would suggest that it is correct.

Reply:
Yes because empty graph will have an empty array as Eulerian trail, in case if a Eulerian trail is not present I am throwing an error. Returning empty array is correct I think, according to the definition in https://en.wikipedia.org/wiki/Eulerian_path.